sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -219,4 +220,41 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("MAX_CASES_BRANCHES") {
+    import testImplicits._
+
+    val original = spark.conf.get(SQLConf.MAX_CASES_BRANCHES)
+    try {
+      withTable("tab1") {
+        spark
+          .range(10)
+          .select('id as 'a, 'id as 'b, 'id as 'c, 'id as 'd)
```

Comment:
I think we only need `a`? or just `spark.range(10).write.saveAsTable`, then we can use `id` in the case when